ykrt/src/compile/jitc_yk/codegen/x86_64.rs

```diff
@@ -226,12 +229,28 @@ impl<'a> X64CodeGen<'a> {
     fn codegen_loadarg_instr(
         &mut self,
         inst_idx: jit_ir::InstrIdx,
-        _inst: &jit_ir::LoadArgInstruction,
+        inst: &jit_ir::LoadArgInstruction,
     ) {
-        // FIXME: LoadArg instructions are not yet specified. This hack is just to satisfy data
-        // flow dependencies during code generation.
-        dynasm!(self.asm ; mov Rq(WR0.code()), 0);
-        self.reg_into_new_local(inst_idx, WR0);
+        // Find the argument register containing the pointer to the live variables struct.
+        let base_reg = ARG_REGS[JITFUNC_LIVEVARS_ARGIDX].code();
+
+        // Now load the value into a new local variable from [base_reg+off].
+        match i32::try_from(inst.off()) {
+            Ok(off) => {
+                let size = inst_idx.instr(self.jit_mod).def_byte_size(self.jit_mod);
+                debug_assert!(size <= REG64_SIZE);
+                dbg!(size);
```

Comment:
`dbg!`

Reply:
Fixed in 515831dfe6b142b88307884947653ec5a5a87122